sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2107,6 +2177,29 @@ class Analyzer(override val catalogManager: CatalogManager)
             }
         }
     }
+
+    /**
+     * Check if the input `fn` implements the given `methodName`. If `inputType` is set, it also
+     * tries to match it against the declared parameter types.
+     */
+    private def findMethod(
+        fn: BoundFunction,
+        methodName: String,
+        inputTypeOpt: Option[Seq[DataType]] = None): Option[Method] = {
+      val cls = fn.getClass
+      inputTypeOpt match {
+        case Some(inputType) =>
+          try {
+            val argClasses = inputType.map(ScalaReflection.dataTypeJavaClass)
+            Some(cls.getDeclaredMethod(methodName, argClasses: _*))
+          } catch {
+            case _: NoSuchMethodException =>
+              None
+          }
+        case None =>
+          cls.getDeclaredMethods.find(_.getName == methodName)
```

Comment:
shall we check if it's a 0-arg method?

Reply:
Hmm sorry not sure what you mean. The case when arg types are known is handled above. This is for `AggregateFunction.update` where we just check if the method is overridden: due to type erasure we can't do an exact match with argument types.